Packs/Phishing/Scripts/CheckEmailAuthenticity/CheckEmailAuthenticity.py

```diff
@@ -109,15 +109,40 @@ def auth_check(spf_data, dkim_data, dmarc_data, override_dict):
     return 'Pass'
 
 
+def get_authentication_value(headers, original_authentication_header):
+    """
+    Handel the case where the authentication header is given under a different header.
+    This header is represented by the 'original_authentication_header' argument.  This can happen when there is an
+     intermediate server which changes the email and holds the original value of the header in a different header.
+     For more info, see issue #46364.
+    Args:
+        headers: The headers dict argument given by the user
+        original_authentication_header: The name of a header which holds the original value of the
+        Authentication-Results header.
+
+    Returns:
+        The suitable authenticator header.
+
+    """
+    headers = [header for header in headers if isinstance(header, dict)]
```

Comment:
I took this logic from the existing logic was in the script's main, i assumed that it was there since there is a case that this happens in.
Adding the filter to the next line